src/main/java/mate/academy/dao/BookDaoImp.java

```diff
@@ -0,0 +1,144 @@
+    private static final String FIND_ALL = "SELECT * FROM books";
+    private static final String DELETE_BY_ID = "DELETE FROM books WHERE id = ?";
+    private static final String ADD_TO_BOOK = "INSERT INTO books (title, price) VALUES (?, ?)";
+    private static final String UPDATE_IN_BOOK = "UPDATE books SET Price = ? WHERE title = ?";
+    private static final String PATH_TO_FILE = "src/main/resources/init_db.sql";
+    private static final String COLUM_LABEL_TITLE = "title";
+    private static final String COLUM_LABEL_PRICE = "price";
+    private static final String COLUM_LABEL_ID = "id";
+
+    @Override
+    public Book create(Book book) {
+        try (PreparedStatement statement = ConnectionUtil.getConnection()
+                .prepareStatement(ADD_TO_BOOK, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new DataProcessingException("Can't create new row "
+                        + "affected rows was = " + affectedRows);
+            }
+            ResultSet resultKey = statement.getGeneratedKeys();
+            if (resultKey.next()) {
+                book.setId(resultKey.getObject(1, Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create new row", e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        Book book = null;
+        try (PreparedStatement statement = ConnectionUtil.getConnection()
+                .prepareStatement(FIND_BY_ID)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                String title = resultSet.getString(COLUM_LABEL_TITLE);
+                BigDecimal price = resultSet.getBigDecimal(COLUM_LABEL_PRICE);
+                book = new Book();
+                book.setPrice(price);
+                book.setTitle(title);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find data by id", e);
+        }
+        return Optional.ofNullable(book);
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> books = new ArrayList<>();
+        try (PreparedStatement statement = ConnectionUtil.getConnection()
+                .prepareStatement(FIND_ALL)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                String title = resultSet.getString(COLUM_LABEL_TITLE);
+                BigDecimal price = resultSet.getBigDecimal(COLUM_LABEL_PRICE);
+                Book book = new Book();
+                book.setPrice(price);
+                book.setTitle(title);
```

Comment:
Try to avoid code duplication. Especially, when you are working with ResultSet. Move retrieving data from ResultSet into Entity object to a separate private method.